src/DynamoRevit/Models/RevitDynamoModel.cs

```diff
@@ -495,56 +505,16 @@ private void OnPythonEngineCollectionChanged(object sender, NotifyCollectionChan
         {
-            if (UnwrapElementMarshaler == null)
-            {
-                UnwrapElementMarshaler = new DataMarshaler();
-                UnwrapElementMarshaler.RegisterMarshaler((Revit.Elements.Element element) => element.InternalElement);
-                UnwrapElementMarshaler.RegisterMarshaler((Category element) => element.InternalCategory);
-                UnwrapElementMarshaler.RegisterMarshaler((Python.Runtime.PyObject pyObj) =>
-                        {
-                            IntPtr gs = Python.Runtime.PythonEngine.AcquireLock();
-                            try
-                            {
-                                using (Python.Runtime.Py.GIL())
-                                {
-                                    if (Python.Runtime.PyDict.IsDictType(pyObj))
-                                    {
-                                        using (var pyDict = new Python.Runtime.PyDict(pyObj))
-                                        {
-                                            var dict = new Python.Runtime.PyDict();
-                                            foreach (Python.Runtime.PyObject item in pyDict.Items())
-                                            {
-                                                dict.SetItem(
-                                                    Python.Runtime.ConverterExtension.ToPython(UnwrapElementMarshaler.Marshal(item.GetItem(0))),
-                                                    Python.Runtime.ConverterExtension.ToPython(UnwrapElementMarshaler.Marshal(item.GetItem(1)))
-                                                );
-                                            }
-                                            return dict;
-                                        }
-                                    }
-                                    var unmarshalled = pyObj.AsManagedObject(typeof(object));
-                                    return UnwrapElementMarshaler.Marshal(unmarshalled);
-                                }
-                            }
-                            catch (Exception e)
-                            {
-                                Logger.Log($"error marshaling python object {pyObj.Handle} {e.Message}");
-                                return pyObj;
-                            }
-                            finally
-                            {
-                                Python.Runtime.PythonEngine.ReleaseLock(gs);
-                            }
-                        });
-            }
-            Func<object, object> unwrap = UnwrapElementMarshaler.Marshal;
             // Turn off element binding during python script execution
             ElementBinder.IsEnabled = false;
-            // register UnwrapElement method
-            scopeSet("UnwrapElement", unwrap);
+            if (UnwrapElementMarshaler != null)
+            {
+                Func<object, object> unwrap = UnwrapElementMarshaler.Marshal;
```

Comment:
Do nested `dict` and `list` objects work appropriately?

Do we know if there were existing workarounds related to python wrapping done by 3rd party packages (nodes) or included in python scripts?

Reply:
I think yes. I've been looking into some graphs from the tests in D4R that make use of that code and it seems to work as expected.